manualdirectdebit.php

```diff
@@ -175,138 +175,21 @@ function manualdirectdebit_civicrm_postProcess($formName, &$form) {
   $action = $form->getAction();
 
   if ($formName == 'CRM_Contribute_Form_Contribution' && $action == CRM_Core_Action::ADD) {
-    $contributionId = (int) $form->getVar('_id');
-    $assignedContactId = (int) $form->getVar('_contactID');
-    $mandateId = getIdOfNewMandate($assignedContactId);
-    if ($form->_params['is_recur'] == 1) {
-      $recurrContributionId = $form->_params['contributionRecurID'];
-      saveDependencyBetweenContributionAndMandate($recurrContributionId, $mandateId);
+    $createManualDirectDebit = new CRM_ManualDirectDebit_Hook_PostProcess_CreateDirectDebitMandate((int) $form->getVar('_contactID'));
```

Comment:
I think this can be rewritten into this : 

```php
if ($formName == 'CRM_Contribute_Form_Contribution' && $action == CRM_Core_Action::ADD) {
$manualDirectDebit = CRM_ManualDirectDebit_Hook_PostProcess_Contribution_DirectDebitMandate($form);

$directDebitMandate->create();
}

now inside the class, the create method will look something like : 

function create() {
$this->mandateID = $this->createMandate();
$this->assignContributionMandate();
$this->assignRecurringContributionMandate();
}

and the check for if ($form->_params['is_recur']) { should be moved to assignRecurringContributionMandate() instead

```